XamarinCommunityToolkit/Views/SegmentedView/SegmentedView.shared.cs

```diff
@@ -0,0 +1,199 @@
+			BindableProperty.Create("Display", typeof(string), typeof(SegmentedView), default(string));
+
+		string GetDisplayMember(object item)
+		{
+			if (ItemDisplayBinding == null)
+				return item.ToString();
+
+			//TODO: Fix this
+			//ItemDisplayBinding.Apply(item, this, displayProperty);
+			//ItemDisplayBinding.Unapply();
+
+			return (string)GetValue(displayProperty);
+		}
+
+		static void OnItemsSourceChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			((SegmentedView)bindable)?.OnItemsSourceChanged((IList)oldValue, (IList)newValue);
+		}
+
+		void OnItemsSourceChanged(IList oldValue, IList newValue)
+		{
+			var oldObservable = oldValue as INotifyCollectionChanged;
+			if (oldObservable != null)
+				oldObservable.CollectionChanged -= CollectionChanged;
+
+			var newObservable = newValue as INotifyCollectionChanged;
+			if (newObservable != null)
+			{
+				newObservable.CollectionChanged += CollectionChanged;
+			}
+
+			if (newValue != null)
+				ResetItems();
+		}
+
+		void CollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+		{
+			switch (e.Action)
+			{
+				case NotifyCollectionChangedAction.Add:
+					AddItems(e);
+					break;
+				case NotifyCollectionChangedAction.Remove:
+					RemoveItems(e);
+					break;
+				default: //Move, Replace, Reset
+					ResetItems();
+					break;
+			}
+		}
+
+		void AddItems(NotifyCollectionChangedEventArgs e)
+		{
+			var index = e.NewStartingIndex < 0 ? Items.Count : e.NewStartingIndex;
+			foreach (var newItem in e.NewItems)
+				((LockableObservableListWrapper)Items).InternalInsert(index++, GetDisplayMember(newItem));
+		}
+
+		void RemoveItems(NotifyCollectionChangedEventArgs e)
+		{
+			var index = e.OldStartingIndex < Items.Count ? e.OldStartingIndex : Items.Count;
+			foreach (var _ in e.OldItems)
+				((LockableObservableListWrapper)Items).InternalRemoveAt(index--);
+		}
+
+		void ResetItems()
+		{
+			if (ItemsSource == null)
+				return;
+
+			((LockableObservableListWrapper)Items).InternalClear();
+
+			foreach (var item in ItemsSource)
+				((LockableObservableListWrapper)Items).InternalAdd(GetDisplayMember(item));
+
+			UpdateSelectedItem(SelectedIndex);
+		}
+
+		static void OnSelectedItemChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var segments = (SegmentedView)bindable;
+			segments.SelectedItem = newValue;
+		}
+
+		void UpdateSelectedItem(int index)
+		{
+			if (index == -1)
+			{
+				SelectedItem = null;
+				return;
+			}
+
+			if (ItemsSource != null)
+			{
+				SelectedItem = ItemsSource[index];
+				return;
+			}
+
+			SelectedItem = Items[index];
+		}
+
+		public int SelectedIndex
+		{
+			get => (int)GetValue(SelectedIndexProperty);
+			set => SetValue(SelectedIndexProperty, value);
+		}
+
+		static void OnSegmentSelected(BindableObject bindable, object oldValue, object newValue)
+		{
+			if (!(bindable is SegmentedView segment))
+				return;
+			int.TryParse(newValue?.ToString(), out var index);
```

Comment:
and if the `TryParse` fails?

Reply:
Fixed it. Defaulting to `0` in case of failure.